astropy/io/votable/tests/test_vo.py

```diff
@@ -251,6 +252,26 @@ def test_ref_table(self):
     def test_iter_coosys(self):
         assert len(list(self.votable.iter_coosys())) == 1
 
+@pytest.mark.parametrize(
+    "columns, expected_missing",
+    [
+        # a single non-existent column
+        (["c1"], ["c1"]),
+        # multiple missing columns (checking that order is preserved)
+        (["c1", "c2", "c3"], ["c1", "c2", "c3"]),
+        # mixing existing with missing columns
+        (["c1", "string_test", "c2"], ["c1", "c2"]),
```

Comment:
Using [`pytest.param()`](https://docs.pytest.org/en/stable/reference/reference.html#pytest-param) instances with a custom `id` is much better than commenting the parameters, see https://docs.pytest.org/en/stable/example/parametrize.html#different-options-for-test-ids and https://docs.pytest.org/en/stable/example/parametrize.html#set-marks-or-test-id-for-individual-parametrized-test

Reply:
I don't fully agree: comments are free form, can be much longer, and live next to the actual examples, ids don't.
In particular, the comment on my second case explains *why* this case is tested, which isn't something that should be described by ids, in my honest opinion.
 However, I suppose I can have short ids *as well* as comments.

Reply:
A good thing to do would be to actually look at the output `pytest` produces. In the following the tests always failed because I did not apply your bugfix.

Without the ids:
```
======================================= short test summary info =======================================
FAILED astropy/io/votable/tests/test_vo.py::test_select_missing_columns[columns0-expected_missing0] - AssertionError: Regex pattern did not match.
FAILED astropy/io/votable/tests/test_vo.py::test_select_missing_columns[columns1-expected_missing1] - AssertionError: Regex pattern did not match.
FAILED astropy/io/votable/tests/test_vo.py::test_select_missing_columns[columns2-expected_missing2] - AssertionError: Regex pattern did not match.
========================================== 3 failed in 0.54s ==========================================
```
With your ids:
```
======================================= short test summary info =======================================
FAILED astropy/io/votable/tests/test_vo.py::test_select_missing_columns[single-missing-col] - AssertionError: Regex pattern did not match.
FAILED astropy/io/votable/tests/test_vo.py::test_select_missing_columns[multiple-missing-cols] - AssertionError: Regex pattern did not match.
FAILED astropy/io/votable/tests/test_vo.py::test_select_missing_columns[not-just-missing-cols] - AssertionError: Regex pattern did not match.
========================================== 3 failed in 0.59s ==========================================
```
But we could also obtain the following:
```
======================================= short test summary info =======================================
FAILED astropy/io/votable/tests/test_vo.py::test_select_missing_columns_error_message[basic] - AssertionError: Regex pattern did not match.
FAILED astropy/io/votable/tests/test_vo.py::test_select_missing_columns_error_message[check-ordering] - AssertionError: Regex pattern did not match.
FAILED astropy/io/votable/tests/test_vo.py::test_select_missing_columns_error_message[list-only-missing] - AssertionError: Regex pattern did not match.
========================================== 3 failed in 0.54s ==========================================
```
To me it is obvious that `FAILED astropy/io/votable/tests/test_vo.py::test_select_missing_columns_error_message[list-only-missing]` is much better than `FAILED astropy/io/votable/tests/test_vo.py::test_select_missing_columns[columns2-expected_missing2]`.

If you still want to have comments _in addition_ to the ids then I don't see any harm in having them.